python/tests/test_integration.py

```diff
@@ -20,44 +20,43 @@ class AsyncMock(mock.MagicMock):
 @pytest.mark.asyncio
 async def test_e2e_flow(tmpdir):
     task_lifecycle_util = TaskLifecycleUtil(Path(tmpdir))
-    app_lifecycle_handler = AsyncMock(spec_set=AppLifecycleHandler)
     requestor_handler = AsyncMock(spec_set=RequestorAppHandler)
     provider_handler = AsyncMock(spec_set=ProviderAppHandler)
 
+    task_struct = structs.Task(
+        env_id='test_env',
+        prerequisites={}
+    )
     task_id = 'test_task_id123'
     max_subtasks_count = 1
     resources = []
     task_params = {'test_param': 'test_value'}
 
+    # Mocks have to be configured before calling init_requestor_with_handler
+    # because it creates copies of these objects when spawning server process.
+    requestor_handler.create_task.return_value = task_struct
+    requestor_handler.has_pending_subtasks.return_value = True
+    requestor_handler.next_subtask.return_value = \
+        structs.Subtask('subtask_id', {}, [])
+    result_path = Path('result')
```

Comment:
Kinda lost in-between all those mocks :)

Reply:
Not anymore.